roles/jobtemplate/tasks/main.yml

```diff
@@ -1,23 +1,38 @@
 ---
+- name: Read Secret Configuration from connection_secret
+  kubernetes.core.k8s_info:
+    kind: Secret
+    api_version: v1
+    name: "{{ connection_secret }}"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+  register: tower_connection_secret
+  when: connection_secret is defined
 
-- name: Read Secret Configuration
+- name: DEPRECATED Read Secret Configuration from tower_auth_secret
   kubernetes.core.k8s_info:
     kind: Secret
-    namespace: '{{ ansible_operator_meta.namespace }}'
+    api_version: v1
     name: "{{ tower_auth_secret }}"
-  register: tower_config_secret
+    namespace: "{{ ansible_operator_meta.namespace }}"
+  register: tower_connection_deprecated_secret
+  when: tower_auth_secret is defined
+
+- name: Set fact based on secret results
+  set_fact:
+     tower_config_secret: "{{ tower_connection_secret }}"
```

Comment:
Only nitpick is that if we are introducing new variables, can we make it `awx`, not `tower` where possible?  Or better yet, remove the prefix all together if we can?

Reply:
Oh, I guess this isn't net new.  Carry on :)